### ISSUE TITLE
Menu Items: unable to create a Menu Type Alias with the same alias of another menu type item

### ISSUE THREAD

**Issue by helvecio** [joomla/joomla-cms#15399]
Joomla has changed the way it handles alias. 

As an example, look at the structure of this site:

* Home
* Collections
    * Black Star
        * Gold
        * Silver
        * SilverGold
* Contact

in the main menu I have the following items:

* Home
* Collections - article
    * Black Star - article with images linking to the sub-items below
* Contact

As my client wanted the main menu to display the first level only, and the article Black Star to have images pointing to the sub-items, I created a hidden menu with the remaining sub-levels to have the URL displayed correctly:

* Collections - alias to the menu item with the same name in the main menu
    * Black Star - alias to the menu item with the same name in the main menu
        * Gold - article
        * Silver - article
        * SilverGold - article

Previously, when creating an alias menu item to another type of menu item, Joomla would have the alias field filled with a number generated with the date and time the article was created. With SEO turned on, then it replaced this number with the alias of the original menu item.

That means that when I clicked on **Collection > Black Star** I would get **www.site.com/collections/black-star.html**

Once I was in the Black Star page, when clicking on the image **Gold**, I would get **www.site.com/collections/black-star/gold.html**

Now, that cannot be achieved anymore, because of the "conflict" of having two different menu items on the same level, although one of them is an alias.

I have to hardcode a different alias, e.g. collections-alias, which gives me a totally different URL and affects the site's SEO.

I heard this was changed because people were having problems with URLs with date and time. 

This was easily fixed by simply deleting the offending alias and re-saving the menu item. When it was recreated, it would show on the live site the correct URL. Sometimes, it was necessary to change the menu item from alias type to another, clear the alias field, then go back to alias type and all worked well.

This change removed the capability of Joomla to create perfect SEO URLs in different scenarios.

In short, it solved a problem that could be fixed by any developer who did a basic research on the subject and replaced it with another that cannot be fixed at all.

**Comment by rdeutz** [comment]
I guess it's for 3.7.0-rc3 or staging you tested this, right?
If yes, do you know the PR with the change you mentioned?

**Comment by brianteeman** [comment]
Not directly related to the issue but it does go to the use case.

@helvecio did you know you do not need to create a hidden menu and aliases. You can do it all in the main menu and set the menu item to be hidden (option - display in menu). Much easier and especially makes things like module assignments easy as you only have one menu item and not two.

**Comment by helvecio** [comment]
Thank you @brianteeman 

I tried to use this feature in 3.6.5 but it doesn't seem to work as you mention. Actually, I was not sure of its purpose until now.

In 3.6.5, when I choose **Link Type > Display Menu > No** the menu item is still displayed.

In 3.7RC it works as expected. 

Thankfully it is about to be released!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15399">issues.joomla.org/tracker/joomla-cms/15399</a>.</sub>

**Comment by zero-24** [comment]
> In 3.7RC it works as expected.

So we can close this issue? @helvecio

**Comment by helvecio** [comment]
Maybe this could be addressed if there are any plans for fixes in 3.6.x.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15399">issues.joomla.org/tracker/joomla-cms/15399</a>.</sub>

**Comment by rdeutz** [comment]
I don't think we will release a 3.6.x when we have 3.7.0 released

**Comment by helvecio** [comment]
Close it then, please.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15399">issues.joomla.org/tracker/joomla-cms/15399</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
closed as written above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15399">issues.joomla.org/tracker/joomla-cms/15399</a>.</sub>
